File: ruby/raindrops/raindrops.rb
class Raindrops

  def self.convert(int)
    rain = ""
    if int % 3 == 0
      rain += "Pling"
    end
    if int % 5 == 0
      rain += "Plang"
    end
    if int % 7 == 0
      rain += "Plong"
    end
    if rain == ""
      return int.to_s
    end
    return rain
  end

end

module BookKeeping
  VERSION = 3
end

File: c/raindrops/src/raindrops.c
#include "raindrops.h"
#include "stdio.h"
#include "string.h"

char *convert(char rain[], int drops)
{
   if (drops % 3 == 0) {
      strcat(rain, "Pling");
   }
   if (drops % 5 == 0) {
      strcat(rain, "Plang");
   }
   if (drops % 7 == 0) {
      strcat(rain, "Plong");
   }
   if (strlen(rain) == 0 ){
     sprintf(rain, "%d", drops);
   }
   return rain;
}
